modules/e2ee/ManagedKeyHandler.js

```diff
@@ -102,21 +106,21 @@ export class ManagedKeyHandler extends KeyHandler {
      * Advances (using ratcheting) the current key when a new participant joins the conference.
      * @private
      */
-    _onParticipantJoined() {
+    async _onParticipantJoined() {
         if (this._conferenceJoined && this.enabled) {
-            this._ratchetKey();
+            await this._ratchetKey();
         }
     }
 
     /**
      * Rotates the current key when a participant leaves the conference.
      * @private
      */
-    _onParticipantLeft(id) {
+    async _onParticipantLeft(id) {
         this.e2eeCtx.cleanup(id);
 
         if (this.enabled) {
-            this._rotateKey();
+            await this._rotateKey();
```

Comment:
Not sure I understand the reasoning behind this whole change. Is it to avoid 2 participant joined / left operations from running concurrently? If so shouldn't we perhaps use some kind of operations queue?

Reply:
For rotating I have not seen the problem described for ratcheting, but I figured some race condition can occur here, as well. Indeed an operation queue would make more sense, but I think we can add this in the state management system we discussed.